File: release/Sport@Unibe/src/ch/unibe/sport/DBAdapter/tables/EventPeriods.java
package ch.unibe.sport.DBAdapter.tables;

import android.content.Context;
import ch.unibe.sport.utils.Utils;

public class EventPeriods extends Table {
	public static final String TAG = EventPeriods.class.getName();
	
	public static final String NAME = "eventPeriods";
	public static final String CREATE = "CREATE TABLE " + NAME + " (" +
			"eid integer NOT NULL,"+
			"periods TINYINT NOT NULL,"+
			"CHECK (periods BETWEEN 1 AND 5),"+
			"CONSTRAINT unique_eid_period UNIQUE (eid,periods)"+
		");";
	
	public static final String[] DB = {
		"eid",						// 0
		"periods",					// 1
	};
	
	public static final int EID = 0, PERIOD= 1;
	
	public EventPeriods(Context context) {
		super(context, NAME, CREATE, DB);
	}
	
	public int[] getPeriods(int eventID){
		db.open(context,TAG);
		int[] result = 
			Utils.getRow(
				Utils.transpose(
					getResultInt(
						db.getDB().query(
							NAME,
							new String[]{DB[PERIOD]}, DB[EID] + " = ?",
							new String[]{""+eventID}, null, null, DB[PERIOD]
						)
					)
				),
			EID);
		db.close(TAG);
		return result;
	}
}

File: release/Webservice/lib/mysql/tables/EventPeriods.php
<?php
/**
 * Created by PhpStorm.
 * User: Aliaksei Syrel
 * Date: 19.11.13
 * Time: 16:25
 */

require_once "Table.php";

class EventPeriods extends Table {
    public static $NAME = "eventPeriods";

    private static $CREATE = "CREATE TABLE eventPeriods (
        eid integer NOT NULL,
		period TINYINT NOT NULL,
		CHECK (period BETWEEN 1 AND 5),
		CONSTRAINT unique_eid_period UNIQUE (eid,period),
		FOREIGN KEY (eid)
        REFERENCES events(eid)
        ON DELETE CASCADE
        ON UPDATE CASCADE
	) ENGINE=InnoDB DEFAULT CHARSET=utf8;";

    public static $DB = array(
        "eid",			    // 0
        "period"		    // 1
    );

    const EID = "eid", PERIOD = "period";

    function __construct() {
        parent::__construct(self::$NAME,self::$CREATE,self::$DB);
    }

    /**
     * Adds period to course
     * @param $eventID - course id
     * @param $period - period number between 1 and 5
     */
    public function add($eventID,$period){
        $this->insert(array(self::EID,self::PERIOD),array($eventID,$period));
    }

    public function getPeriods($eventID){
        return $this->selectByID(array(self::EID),array($eventID),array(self::PERIOD));
    }

    public function updateAll($eventID, $periods){
        $this->deleteByID(array(self::EID),array($eventID));
        foreach($periods as $period){
            $this->add($eventID,$period);
        }
    }

} 